### PR TITLE
chore: added figma design and badges addons to storybook

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

Badge for deprecated components. With this we can control tags like Migrated, Spectrum-two etc on component level

Screenshot 2024-04-19 at 5 46 27 PM

Design Tab
Screenshot 2024-04-19 at 5 19 26 PM

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
This seems like a cool addition, however, this is not my favorite: 
IMG_6689

Maybe we could work with the design team on an open source entry point to these design files?

**Comment by Rajdeepc** [comment]
> This seems like a cool addition, however, this is not my favorite: IMG_6689
> 
> Maybe we could work with the design team on an open source entry point to these design files?

Yes you are right. I will talk to design and see if I find a public url without auth.

**Comment by castastrophe** [comment]
In the short term, you can toggle add-ons based on environment variable so you could show the design add-on only for dev mode.

**Comment by Rajdeepc** [comment]
> * Storybook

Yes for the time being I have added this addon for `dev mode` only but definitely we should add this as a public url in future.

**Comment by Rajdeepc** [comment]
@Westbrook does this make sense to include this in the current state of main branch considering we will be deleting the deprecated items in SWC 1.0 release?

File: projects/css-custom-vars-viewer/package.json [hunk 1]
```diff
@@ -48,9 +48,11 @@
     },
     "devDependencies": {
         "@custom-elements-manifest/analyzer": "^0.9.0",
+        "@geometricpanda/storybook-addon-badges": "^2.0.2",
         "@open-wc/eslint-config": "^12.0.0",
         "@open-wc/testing": "^3.2.0",
         "@storybook/addon-a11y": "^7.5.0",
+        "@storybook/addon-designs": "^8.0.0",
```
Comment (Westbrook) [inline projects/css-custom-vars-viewer/package.json:55]: Why here?

File: yarn.lock [hunk 2]
```diff
@@ -5386,9 +5411,83 @@
   resolved "https://registry.yarnpkg.com/@spectrum-css/vars/-/vars-9.0.8.tgz#6af3bcdace903b8461f5fcd4c9aa23e70128a456"
   integrity sha512-rGfd7jqXOdR69bEjrRP58ynuIeJU0czPfwQvzhtCzg7jKVukV+efNHqrs086sC6xutB3W4TF71K/dZMr3oyTyg==
 
+"@spectrum-web-components/base@^0.40.5":
```
Comment (Westbrook) [inline yarn.lock:5414]: Why these changes?

**Review by Westbrook — COMMENTED** [review]
Do we also want to do `iconset` here?

I think with this cauterized to dev time, we can move forward with this. Couple of notes internally and then we can land it.

File: yarn.lock (hunk 2 above)
Comment (Rajdeepc) [reply yarn.lock]: These should be transitive depedencies. If you run a yarn install on `main` branch you can see these changes

Comment (Westbrook) [reply yarn.lock]: I don't think they belong. Do you have a `react` directory locally?
